src/main/kotlin/app/revanced/patcher/apk/Apk.kt

```diff
@@ -5,469 +5,253 @@ package app.revanced.patcher.apk
+    private val openFiles = mutableSetOf<String>()
 
-            /**
-             * Write the resources for [Apk.Split.Library].
-             *
-             * @param resources Will be ignored.
-             * @param patchApk The [Apk] file to write the resources to.
-             * @param apkWorkDirectory The directory where the resources are stored.
-             * @param metaInfo Will be ignored.
-             */
-            override fun writeResources(
-                resources: AndrolibResources, patchApk: File, apkWorkDirectory: File, metaInfo: MetaInfo
-            ) {
-                // Do not compress libraries (.so) for speed, because the patchApk is a temporal file.
-                ZipUtils.zipFolders(apkWorkDirectory, patchApk, null, listOf("so"))
-
-                // Write the patchApk file containing the manifest file.
-                apkWorkDirectory.resolve(patchApk.name).also { manifestPatchApk ->
-                    super.writeResources(resources, manifestPatchApk, apkWorkDirectory, metaInfo)
-                }.let { manifestPatchApk ->
-                    // Copy AndroidManifest.xml from manifestPatchApk to patchApk.
-                    fun File.createFs() = FileSystems.newFileSystem(toPath(), null as ClassLoader?)
-                    manifestPatchApk.createFs().use { manifestPatchApkFs ->
-                        patchApk.createFs().use { patchApkFs ->
-                            // Delete AndroidManifest.xml from patchApk and copy it from manifestPatchApk.
-                            patchApkFs.getPath("/AndroidManifest.xml").also { manifestPath ->
-                                patchApkFs.provider().delete(manifestPath)
-                                manifestPatchApkFs.getPath("/AndroidManifest.xml").copyTo(manifestPath)
-                            }
-                        }
-                    }
-                }
-            }
-
-            /**
-             * Read resources for an [Apk] file.
-             *
-             * @param androlib The [Androlib] instance to decode the resources with.
-             * @param extInputFile The [Apk] file.
-             * @param outDir The directory to write the resources to.
-             * @param resourceTable The [ResTable] to use.
-             */
-            override fun readResources(
-                androlib: Androlib, extInputFile: ExtFile, outDir: File, resourceTable: ResTable?
-            ) {
-                // Decode the manifest without looking up attribute references because there is no resources.arsc file.
-                androlib.decodeManifestFull(extInputFile, outDir, resourceTable)
-                androlib.decodeRawFiles(extInputFile, outDir, ApkDecoder.DECODE_ASSETS_NONE)
-            }
+    internal fun lockFile(path: String) {
+        if (openFiles.contains(path)) {
+            throw ApkException.Decode("Path \"$path\" is locked. If you are a patch developer, make sure you always close files.")
```

Comment:
Why is locking mechanism implemented in this class?

Reply:
A locking mechanism is required because not having it may lead to hard-to-diagnose bugs in patches. Changes might be unintentionally discarded if you have multiple instances of a single file.

I just realized the locking mechanism also has problems with resource file resolution that I will have to deal with.

The locking mechanism should definitely be moved to the Archive class. The `DomFileEditor` locking can probably also be removed.

Reply:
The duplicate locking mechanism has been removed. Marking this as resolved.